waku/waku_archive/driver/builder.nim

```diff
@@ -78,22 +80,33 @@ proc new*(T: type ArchiveDriver,
     return ok(res.get())
 
   of "postgres":
-    const MaxNumConns = 5 #TODO: we may need to set that from app args (maybe?)
-    let res = PostgresDriver.new(url, MaxNumConns, onErrAction)
-    if res.isErr():
-      return err("failed to init postgres archive driver: " & res.error)
-
-    let driver = res.get()
-
-    try:
-      # The table should exist beforehand.
-      let newTableRes = waitFor driver.createMessageTable()
-      if newTableRes.isErr():
-        return err("error creating table: " & newTableRes.error)
-    except CatchableError:
-      return err("exception creating table: " & getCurrentExceptionMsg())
-
-    return ok(driver)
+    when defined(postgres):
+      const MaxNumConns = 5 #TODO: we may need to set that from app args (maybe?)
+      let res = PostgresDriver.new(url, MaxNumConns, onErrAction)
+      if res.isErr():
+        return err("failed to init postgres archive driver: " & res.error)
+
+      let driver = res.get()
+
+      try:
+        # The table should exist beforehand.
+        let newTableRes = waitFor driver.createMessageTable()
+        if newTableRes.isErr():
+          return err("error creating table: " & newTableRes.error)
+      except CatchableError:
+        return err("exception creating table: " & getCurrentExceptionMsg())
+
+      return ok(driver)
+
+    else:
+      ## Error variable to be used if the user is trying to mount Store with a Postgres database
+      var errPgMsg = "ERROR:"
```

Comment:
Ah, the error message here (i.e. the `builder.nim`) can have no knowledge of the encapsulating application, `wakunode2`, or `make` commands, etc. It's good that we're trying to be helpful, but at most the driver knows something like: "Postgres has been configured but not been compiled. Check compiler definitions."
I think it's OK to link to the install guide here, even though strictly speaking this module should not even know it's part of nwaku. :)

(Not for this PR, but the correct way to do it is to enumerate the error types here, rather than just returning error strings. That way the application `wakunode2` receives an error result when setting up the archive and the application can then log more instructions as it has the context to do so. For now, this will be overkill.)